CMakeLists.txt

```diff
@@ -74,6 +74,7 @@ option(${PROJECT_NAME}_TESTS "Build the tests" OFF)
 option(${PROJECT_NAME}_EXAMPLES "Build the examples" ON)
 option(${PROJECT_NAME}_DOCS "Build the API documentation" OFF)
 option(${PROJECT_NAME}_QT6 "Build against Qt 6" ON)
+option(${PROJECT_NAME}_WERROR "Treat build warnings as errors" OFF)
```

Comment:
CMake provides a builtin option for that already: CMAKE_COMPILE_WARNING_AS_ERROR

I learned that very recently in https://codereview.kdab.com/c/training-material/+/151437

Should we just document to use that instead?

Reply:
good idea

Reply:
the preset serves as documentation